src/Raven.Studio/typescript/components/common/shell/databasesSlice.ts

```diff
@@ -119,27 +123,82 @@ export const selectDatabasesSummary = (store: RootState) => {
             default:
                 assertUnreachable(state);
         }
+
+        if (db.sharded) {
+            sharded++;
+        } else {
+            nonSharded++;
+        }
+
+        if (db.currentNode.relevant) {
+            local++;
+        } else {
+            remote++;
+        }
     });
 
-    return {
-        count: allDatabases.length,
-        loading,
-        error,
-        offline,
-        disabled,
-        online,
-    };
+    return [
+        { value: "Online", label: "Online", count: online },
+        { value: "Offline", label: "Offline", count: offline },
+        { value: "Error", label: "Errored", count: error },
+        { value: "Disabled", label: "Disabled", count: disabled },
+        { value: "Sharded", label: "Sharded", count: sharded, verticalSeparatorLine: true },
+        { value: "NonSharded", label: "Non Sharded", count: nonSharded },
+        {
+            value: "Local",
+            label: `Local (Node ${store.databases.localNodeTag})`,
+            count: local,
+            verticalSeparatorLine: true,
+        },
+        { value: "Remote", label: "Remote", count: remote },
+    ];
+};
+
+const isDatabaseInFilterState = (
+    store: RootState,
+    db: DatabaseSharedInfo,
+    filterStates: DatabaseFilterByStateOption[]
+): boolean => {
+    const perNodeState = selectDatabaseState(db.name)(store);
+    const databaseState = DatabaseUtils.getDatabaseState(db, perNodeState);
+
+    if (
```

Comment:
no need for if/return true/return false

just return. 

Optionally we might split that into separate conditions for readability

Reply:
maybe I will improve readability with formating and use // prettier-ignore

Reply:
ok let's leave it